nipype/interfaces/niftyreg/base.py

```diff
@@ -0,0 +1,127 @@
+from distutils.version import StrictVersion
+import os
+import subprocess
+from warnings import warn
+
+from ..base import CommandLine, isdefined, CommandLineInputSpec, traits
+from ...utils.filemanip import split_filename
+
+
+def get_custom_path(command):
+    try:
+        specific_dir = os.environ['NIFTYREGDIR']
+        command = os.path.join(specific_dir, command)
+        return command
+    except KeyError:
+        return command
+
+
+def no_niftyreg(cmd='reg_f3d'):
+    if True in [os.path.isfile(os.path.join(path, cmd)) and
+                os.access(os.path.join(path, cmd), os.X_OK)
+                for path in os.environ["PATH"].split(os.pathsep)]:
+        return False
+    return True
+
+
+class NiftyRegCommandInputSpec(CommandLineInputSpec):
+    """Input Spec for niftyreg interfaces."""
+    # Set the number of omp thread to use
+    omp_core_val = traits.Int(desc='Number of openmp thread to use',
+                              argstr='-omp %i')
+
+
+class NiftyRegCommand(CommandLine):
+    """
+    Base support interface for NiftyReg commands.
+    """
+    _suffix = '_nr'
+    _min_version = '1.5.30'
+
+    def __init__(self, required_version=None, **inputs):
+        super(NiftyRegCommand, self).__init__(**inputs)
+        self.required_version = required_version
+        _version = self.get_version()
+        if _version:
+            _version = _version.decode("utf-8")
+            if StrictVersion(_version) < StrictVersion(self._min_version):
+                msg = 'A later version of Niftyreg is required (%s < %s)'
+                warn(msg % (_version, self._min_version))
+            if required_version is not None:
+                if StrictVersion(_version) != StrictVersion(required_version):
+                    msg = 'The version of NiftyReg differs from the required'
+                    msg += '(%s != %s)'
+                    warn(msg % (_version, self.required_version))
+
+    def check_version(self):
+        _version = self.get_version()
+        if not _version:
+            raise Exception('Niftyreg not found')
+        # Decoding to string:
+        _version = _version.decode("utf-8")
+        if StrictVersion(_version) < StrictVersion(self._min_version):
+            err = 'A later version of Niftyreg is required (%s < %s)'
+            raise ValueError(err % (_version, self._min_version))
+        if self.required_version:
+            if StrictVersion(_version) != StrictVersion(self.required_version):
+                err = 'The version of NiftyReg differs from the required'
+                err += '(%s != %s)'
+                raise ValueError(err % (_version, self.required_version))
+
+    def get_version(self):
+        if no_niftyreg(cmd=self.cmd):
+            return None
+        exec_cmd = ''.join((self.cmd, ' -v'))
+        return subprocess.check_output(exec_cmd, shell=True).strip()
+
+    @property
+    def version(self):
+        return self.get_version()
+
+    def exists(self):
```

Comment:
Hi, no problem.

I am using flake8 but it does not complain with this one. I will edit it.

Cheers,